### PR TITLE
67 button with right icon

### PR DESCRIPTION
**Pull request by Sofiya-kumar**
Changes for Buttons

image

image

image

### CODE REVIEW

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/Button/CustomButton.xaml:1]: Do we need this class? If yes, can you rename the class to something more meaningful?

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+<ContentView xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="Trimble.Modus.Components.Controls.Button.CustomButton"
+             x:Name="customButton"
+             xmlns:constants="clr-namespace:Trimble.Modus.Components.Contant"
+             xmlns:converters="clr-namespace:Trimble.Modus.Components.Converters">
+  <ContentView.Resources>
+    <ResourceDictionary>
+      <converters:ImageNullOrEmptyBoolConverter x:Key="ImageNullOrEmptyBoolConverter" />
+      <converters:InverseBoolConverter x:Key="InverseBoolConverter" />
```
Comment (rthanga1) [inline Trimble.Modus.Components/Controls/Button/CustomButton.xaml:11]: We should be using MAUI toolkit's converter whenever possible. If you can't use them right now, can you create a tech debt issue and link it here?

File: Trimble.Modus.Components/Helpers/AppDataHelper.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Trimble.Modus.Components.Helpers
+{
+    public class AppDataHelper
+    {
+        public static T ParseEnum<T>(string value)
```
Comment (rthanga1) [inline Trimble.Modus.Components/Helpers/AppDataHelper.cs:5]: Can you add comment to this method about what this do?

File: DemoApp/DemoApp/Views/TMButtonPage.xaml [hunk 4]
```diff
@@ -1,127 +1,195 @@
 <?xml version="1.0" encoding="utf-8" ?>
-<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
-             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
-             xmlns:Modus="clr-namespace:Trimble.Modus.Components;assembly=Trimble.Modus.Components"
-             x:Class="DemoApp.TMButtonPage"
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:Modus="clr-namespace:Trimble.Modus.Components;assembly=Trimble.Modus.Components"
+             xmlns:controls="clr-namespace:Trimble.Modus.Components.Controls.Button;assembly=Trimble.Modus.Components"
```
Comment (rthanga1) [inline DemoApp/DemoApp/Views/TMButtonPage.xaml:5]: In the demo app, can we add one radio button with title 'Show icon' with the following options?
'Left', 'Right' 'Both'

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml (hunk 2 above)
Comment (Sofiya-kumar) [reply Trimble.Modus.Components/Controls/Button/CustomButton.xaml:11]: Tried to use the MAUI toolkit. Faced some reference issues. Will create a ticket for this.

File: Trimble.Modus.Components/Helpers/AppDataHelper.cs (hunk 3 above)
Comment (Sofiya-kumar) [reply Trimble.Modus.Components/Helpers/AppDataHelper.cs:5]: Comments added

File: DemoApp/DemoApp/Views/TMButtonPage.xaml (hunk 4 above)
Comment (Sofiya-kumar) [reply DemoApp/DemoApp/Views/TMButtonPage.xaml:5]: Added options to switch the image positons

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml (hunk 1 above)
Comment (Sofiya-kumar) [reply Trimble.Modus.Components/Controls/Button/CustomButton.xaml:1]: Have to remove the TMButton. Once it's removed have to rename this class to TMButton. TMbutton has some reference in modals and toaster

Comment (rthanga1) [reply Trimble.Modus.Components/Controls/Button/CustomButton.xaml:1]: Can you remove that as part of this PR? Or can you create a separate ticket for that to do in the next sprint?

**Review by BAL4-R — DISMISSED** [review]

File: Trimble.Modus.Components/Controls/Button/CustomButton.xaml (hunk 2 above)
Comment (Sofiya-kumar) [reply Trimble.Modus.Components/Controls/Button/CustomButton.xaml:11]: https://github.com/trimble-oss/modus-mobile-maui-components/issues/113

MAUI community toolkit integration

**Review by BAL4-R — APPROVED** [review]

**Pull request merged**